Improves synthesis doc. (#507)
newrelic/entity-definitions: docs/synthesis.md

### Tags

You can also define a set of attributes that can be copied into entity tags.

> These are global tags, and they are applied for the rule or every rule from a set of rules.

Reviewer: ```suggestion
These are global tags, and they are applied for all the rules from the set of rules.
```

Author: Just to notice, maybe I need to rephrase it, but I would like to remark two approaches.
1. There's a unique rule, where the rule itself lives inherited and directly under "synthesis:".
2. There're multiple rules, where they live under "rules:".

Author: Improved, I think it's much clearer like this.

Notice these tag definitions live under ```synthesis```.
```yaml
synthesis:
  tags:
    aws.az:
```

If the attribute `aws.az` is present in the data point that matched any rule, its value will be copied to an entity's tag named `aws.az`.

> And also there are specific tags for every rule. These attributes will only be copied after a datapoint matches a rule, and

Reviewer: ```suggestion
Also, there specific tags for each rule can be defined. These attributes will only be copied after a datapoint matches a rule.
```

> they take priority over the global tags, meaning **if the same tag is defined on both structures**, the specific tag  

Reviewer: ```suggestion
these rule specific tags take priority over the global tags, meaning **if the same tag is defined on both structures**, the rule specific tag  
```
